#! /bin/sh -e

# All lines beginning with `# DPATCH:' are a description of the patch.
# DP: Change link spec for sh4a target to 'sh4alelf_linux' according
# DP: to linker emulation target, which sets default pagesize to 8k.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p1 < $0
        #cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p1 < $0
        #rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- src/gcc/config/sh/elf.h.orig	2007-06-14 08:06:09.000000000 +0000
+++ src/gcc/config/sh/elf.h	2007-06-14 08:09:21.000000000 +0000
@@ -62,7 +62,11 @@
 #define LINK_SPEC SH_LINK_SPEC
 #undef LINK_EMUL_PREFIX
 #if TARGET_ENDIAN_DEFAULT == MASK_LITTLE_ENDIAN
+#ifdef SUPPORT_SH4A
+#define LINK_EMUL_PREFIX "sh4a%{!mb:l}elf"
+#else
 #define LINK_EMUL_PREFIX "sh%{!mb:l}elf"
+#endif /* SUPPORT_SH4A */
 #else
 #define LINK_EMUL_PREFIX "sh%{ml:l}elf"
 #endif

